src/components/SearchBar.tsx

```diff
@@ -1,86 +1,98 @@
-    const itemDate = new Date(dateStr);
-    const before = beforeDate ? new Date(beforeDate) : null;
-    const after = afterDate ? new Date(afterDate) : null;
+  // const isDateInRange = (
+  //   dateStr: string,
+  //   beforeDate: string,
+  //   afterDate: string,
+  // ) => {
+  //   const itemDate = new Date(dateStr);
+  //   const before = beforeDate ? new Date(beforeDate) : null;
+  //   const after = afterDate ? new Date(afterDate) : null;
 
-    if (before && itemDate > before) return false;
-    if (after && itemDate < after) return false;
+  //   if (before && itemDate > before) return false;
+  //   if (after && itemDate < after) return false;
 
-    return true;
-  };
+  //   return true;
+  // };
 
   // Filter logic
-  const handleSearch = () => {
-    const filtered = data.filter((item) => {
-      // Match search term in multiple fields
-      const matchesSearch =
-        item.mcap_file_name.toLowerCase().includes(searchTerm.toLowerCase()) ||
-        item.matlab_file_name
-          .toLowerCase()
-          .includes(searchTerm.toLowerCase()) ||
-        item.notes?.toLowerCase().includes(searchTerm.toLowerCase());
-
-      // Match filters
-      const matchesLocation =
-        filters.location === "" ||
-        item.location.toLowerCase() === filters.location.toLowerCase();
-      const matchesEventType =
-        filters.eventType.toLowerCase() === "" ||
-        item.event_type?.toLowerCase() === filters.eventType.toLowerCase();
-      const matchesDate = isDateInRange(
-        item.date,
-        filters.beforeDate,
-        filters.afterDate,
-      );
-
-      return (
-        matchesSearch && matchesLocation && matchesEventType && matchesDate
-      );
-    });
-    setFilteredData(filtered);
-  };
+  // const handleSearch = () => {
+  //   const filtered = data.filter((item) => {
+  //     // Match search term in multiple fields
+  //     const matchesSearch =
+  //       item.mcap_file_name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+  //       item.matlab_file_name
+  //         .toLowerCase()
+  //         .includes(searchTerm.toLowerCase()) ||
+  //       item.notes?.toLowerCase().includes(searchTerm.toLowerCase());
+
+  //     // Match filters
+  //     const matchesLocation =
+  //       filters.location === "" ||
+  //       item.location.toLowerCase() === filters.location.toLowerCase();
+  //     const matchesEventType =
+  //       filters.eventType.toLowerCase() === "" ||
+  //       item.event_type?.toLowerCase() === filters.eventType.toLowerCase();
+  //     const matchesDate = isDateInRange(
+  //       item.date,
+  //       filters.beforeDate,
+  //       filters.afterDate,
+  //     );
+
+  //     return (
+  //       matchesSearch && matchesLocation && matchesEventType && matchesDate
+  //     );
+  //   });
+  //   setFilteredData(filtered);
+  // };
 
   // Trigger search on filter or search term change
-  useEffect(() => {
-    handleSearch();
-  }, [searchTerm, filters]);
```

Comment:
handleSearch()/sending a request shouldn't change every time a character is added to the `searchTerm`. generally, for cases where you want a "live search" on type like this, you want to debounce the input. feel free to do your own research on this.

however, i think it would just be more practical and simple to just have a submit button but this is a discussion to have with the leads @aesteri @SreeDan

Reply:
Sree's said in the past we should add a "search" button so we're not constantly pinging the server